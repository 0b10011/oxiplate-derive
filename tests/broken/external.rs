use oxiplate_derive::Oxiplate;

#[derive(Oxiplate)]
#[oxiplate = "external.html.oxip"]
struct Data {
    title: &'static str,
}

fn main() {
    let data = Data {
        title: "Hello world",
    };

    print!("{}", data);
}
