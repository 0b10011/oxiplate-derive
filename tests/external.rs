use oxiplate_derive::Oxiplate;

#[derive(Oxiplate)]
#[oxiplate = "external.html.oxip"]
struct AbsoluteData {
    title: &'static str,
    message: &'static str,
}

#[test]
fn absolute() {
    let data = AbsoluteData {
        title: "Oxiplate Example",
        message: "Hello world!",
    };

    assert_eq!(
        format!("{}", data),
        "<h1>Oxiplate Example</h1>\n<p>Hello world!</p>\n"
    );
}

#[test]
fn absolute_2() {
    let data = AbsoluteData {
        title: "Oxiplate Example #2",
        message: "Goodbye world!",
    };

    assert_eq!(
        format!("{}", data),
        "<h1>Oxiplate Example #2</h1>\n<p>Goodbye world!</p>\n"
    );
}
