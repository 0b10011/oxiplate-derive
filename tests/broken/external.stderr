error[E0609]: no field `message` on type `&Data`
 --> templates/external.html.oxip
  |
  | <p>{{ message }}</p>
  |       ^^^^^^^ unknown field
  |
  = note: available field is: `title`
